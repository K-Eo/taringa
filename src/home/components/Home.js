import React from 'react'

import Ad from '../../ads/components/Ad'
import Feed from '../../feed/components/Feed'
import Filterable from '../../filters/components/Filterable'

const filters = {
  hot: {
    displayName: 'Destacados',
    exact: true,
    id: 'hot',
    pathname: '/',
    url: '/feed',
  },
  recents: {
    displayName: 'Recientes',
    exact: false,
    id: 'recents',
    pathname: '/recents',
    url: '/feed?sort=recent',
  },
}

const Home = ({ match }) => {
  const filter = match.params.filter || 'hot'
  const url = filters[filter].url

  return (
    <div className="row">
      <Filterable filters={filters} />

      <div className="col-8">
        <Feed feedId="mi" filter={filter} url={url} />
      </div>

      <div className="col-4">
        <Ad />
      </div>
    </div>
  )
}

Home.defaultProps = {
  match: { params: {} },
}

export default Home
