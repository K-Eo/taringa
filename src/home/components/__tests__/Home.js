import React from 'react'

import { Home } from '../Home'

describe('Home', () => {
  it('exists', () => {
    expect(Home).toBeDefined()
  })

  it('renders', () => {
    const wrapper = shallow(<Home />)

    expect(wrapper).toMatchSnapshot()
  })
})
