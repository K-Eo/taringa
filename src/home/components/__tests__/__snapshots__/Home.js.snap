// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<div
  className="row"
>
  <Connect(Filterable)
    filters={
      Object {
        "hot": Object {
          "displayName": "Destacados",
          "exact": true,
          "id": "hot",
          "pathname": "/",
          "url": "/feed",
        },
        "recents": Object {
          "displayName": "Recientes",
          "exact": false,
          "id": "recents",
          "pathname": "/recents",
          "url": "/feed?sort=recent",
        },
      }
    }
  />
  <div
    className="col-8"
  >
    <Connect(Feed)
      feedId="mi"
      filter="hot"
      url="/feed"
    />
  </div>
  <div
    className="col-4"
  >
    <Ad />
  </div>
</div>
`;
